logger.go

```diff
@@ -0,0 +1,175 @@
+func (l Logger) InfoWithDepth(extraDepth int, args ...interface{}) {
+	l.printWithDepth(infoLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Infoln is equivalent to the global Infoln function, with the addition of prefix and data content from this Logger.
+func (l Logger) Infoln(args ...interface{}) {
+	l.println(infoLog, l.extendWithPfx(args)...)
+}
+
+// InfolnWithDepth is equivalent to the global InfolnWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) InfolnWithDepth(extraDepth int, args ...interface{}) {
+	l.printlnWithDepth(infoLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Infof is equivalent to the global Infof function, with the addition of prefix and data content from this Logger.
+func (l Logger) Infof(format string, args ...interface{}) {
+	l.printf(infoLog, l.pfx(format), l.extend(args)...)
+}
+
+// InfofWithDepth is equivalent to the global InfofWithSepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) InfofWithDepth(extraDepth int, format string, args ...interface{}) {
+	l.printfWithDepth(infoLog, extraDepth, l.pfx(format), l.extend(args)...)
+}
+
+// Warning is equivalent to the global Warning function, with the addition of prefix and data content from this Logger.
+func (l Logger) Warning(args ...interface{}) {
+	l.print(warningLog, l.extendWithPfx(args)...)
+}
+
+// WarningWithDepth is equivalent to the global WarningWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) WarningWithDepth(extraDepth int, args ...interface{}) {
+	l.printWithDepth(warningLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Warningln is equivalent to the global Warningln function, with the addition of prefix and data content from this Logger.
+func (l Logger) Warningln(args ...interface{}) {
+	l.println(warningLog, l.extendWithPfx(args)...)
+}
+
+// WarninglnWithDepth is equivalent to the global WarninglnWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) WarninglnWithDepth(extraDepth int, args ...interface{}) {
+	l.printlnWithDepth(warningLog, extraDepth, l.extendWithPfx(args)...)
+}
+
+// Warningf is equivalent to the global Warningf function, with the addition of prefix and data content from this Logger.
+func (l Logger) Warningf(format string, args ...interface{}) {
+	l.printf(warningLog, l.pfx(format), l.extend(args)...)
+}
+
+// WarningfWithDepth is equivalent to the global WarningfWithDepth function, with the addition of prefix and data content from this Logger.
+func (l Logger) WarningfWithDepth(extraDepth int, format string, args ...interface{}) {
+	l.printfWithDepth(warningLog, extraDepth, l.pfx(format), l.extend(args)...)
+}
+
+// Error is equivalent to the global Error function, with the addition of prefix and data content from this Logger.
+func (l Logger) Error(args ...interface{}) {
```

Comment:
What about ErrorIf?

Reply:
Does that already exist in glog?

Reply:
Yes. Very useful, actually, if the error is not fatal and you can proceed.
err := doSomething()
glog.Errorif(err)
doSomethingElse()

Also a trick I've learned from pranjal: 
defer glog.Errorif(close())

Reply:
[missed?]